descartes_core/include/descartes_core/trajectory_id.h

```diff
@@ -5,6 +5,16 @@
 
 #include <boost/thread/mutex.hpp>
 
+#ifdef WIN32
```

Comment:
By convention, usually you define something like below:
(http://wiki.ros.org/win_ros/Contributing/Dll%20Exports)

```
#ifdef ROS_BUILD_SHARED_LIBS // ros is being built around shared libraries
  #ifdef bondcpp_EXPORTS // we are building a shared lib/dll
    #define BONDCPP_DECL ROS_HELPER_EXPORT
  #else // we are using shared lib/dll
    #define BONDCPP_DECL ROS_HELPER_IMPORT
  #endif
#else // ros is being built around static libraries
  #define BONDCPP_DECL
#endif
```